axelor-sale/src/main/resources/domains/SaleOrder.xml

```diff
@@ -95,6 +95,8 @@
 
     <boolean name="template" title="Template" hidden="true"/>
     <integer name="versionNumber" title="Version Number" readonly="true" default="1"/>
+    <one-to-many name="oldVersionSaleOrderLineList"
+      ref="com.axelor.apps.sale.db.SaleOrderLine" mappedBy="oldVersionSaleOrder" orderBy="sequence"/>
```

Comment:
Please remove this `one-to-many` as we could have too many historic lines, slowing down processes. Instead we will use a query on the field `oldVersionSaleOrder` with the correct filters to not fetch every lines.